src/Tasks/Microsoft.Common.CurrentVersion.targets

```diff
@@ -1863,9 +1863,12 @@ Copyright (C) Microsoft Corporation. All rights reserved.
         <UndefineProperties>%(AnnotatedProjects.UndefineProperties);TargetFramework</UndefineProperties>
       </AnnotatedProjects>
 
-      <!-- If the project is RID agnostic, undefine the RuntimeIdentifier property to avoid another evaluation. -->
-      <AnnotatedProjects Condition="'@(AnnotatedProjects)' == '%(Identity)' and '%(AnnotatedProjects.IsRidAgnostic)' == 'true'">
-        <UndefineProperties>%(AnnotatedProjects.UndefineProperties);RuntimeIdentifier</UndefineProperties>
+      <!-- Add RuntimeIdentifier and SelfContained to the list of global properties that should not flow to the referenced project,
+           unless the project is expecting those properties to flow.
+           We include SelfContained together with RuntimeIdentifier, because otherwise a referenced project could try to build with
+           SelfContained set to true but no RuntimeIdentifier set, which causes an error (for Exe projects). -->
+      <AnnotatedProjects Condition="'@(AnnotatedProjects)' == '%(Identity)' and '%(AnnotatedProjects.IsRidAgnostic)' != 'false'">
```

Comment:
When is '@(AnnotatedProjects)' != '%(Identity)'?

Reply:
I think you have to do this when updating metadata this way, otherwise each update will apply to all the items, or something like that.  @rainersigwald probably knows the details.  Anyway my changes don't change that pattern here, they just change the other conditions.